pkg/kubelet/kubelet.go

```diff
@@ -1687,7 +1687,39 @@ func (kl *Kubelet) syncPod(o syncPodOptions) error {
 	pullSecrets := kl.getPullSecretsForPod(pod)
 
 	// Call the container runtime's SyncPod callback
-	result := kl.containerRuntime.SyncPod(pod, podStatus, pullSecrets, kl.backOff)
+	result, podIPs, sandboxID := kl.containerRuntime.SyncPodSandbox(pod, podStatus, pullSecrets, kl.backOff)
+	kl.reasonCache.Update(pod.UID, result)
+	if err := result.Error(); err != nil {
+		// Do not return error if the only failures were pods in backoff
+		for _, r := range result.SyncResults {
+			if r.Error != kubecontainer.ErrCrashLoopBackOff && r.Error != images.ErrImagePullBackOff {
+				// Do not record an event here, as we keep all event logging for sync pod failures
+				// local to container runtime so we get better errors
+				return err
+			}
+		}
+
+		return nil
+	}
+	if len(podIPs) != 0 {
+		klog.V(4).Infof("pod ip of sandbox is: %v", podIPs[0])
+		_, err := kl.containerRuntime.GetPodStatus(pod.UID, pod.Name, pod.Namespace)
```

Comment:
Why aren't we using the output of GetPodStatus?

Reply:
It is used now.